internal/ziputils/ziputils.go

```
@@ -37,48 +57,37 @@ func ProcessZipFile(zipPath string) ([]string, error) {
+	// Second pass: Extract JSON/.dr files flat, by basename (we've already validated the extensions)
 	for _, f := range reader.File {
-		// Skip directories
-		if f.FileInfo().IsDir() {
+		if f.FileInfo().IsDir() || isIgnoredZipEntry(f.Name) {
 			continue
 		}
 
+		// Guard against basename collisions across directories, since we flatten the hierarchy
+		base := path.Base(f.Name)
+		if prev, ok := seenBasenames[base]; ok {
+			os.RemoveAll(tempDir)
```

golangci-lint (2.12.2)

[error] Error return value of `os.RemoveAll` is not checked

(errcheck)